### DISCUSSION TITLE
[question] How to add other files in build context? e.g. `.npmrc`

### DISCUSSION THREAD

**Discussion by jizusun** [coder/coder#8552] [General] [answered]
I tried to push with `coder template push` but `.npmrc` was skipped.
Also I failed to create the file in the UI, and the error message.

> This extension is not allowed. You only can create files with the following extensions: tf, md, Dockerfile.

![image](https://github.com/coder/coder/assets/4011348/2d0e2fa5-d0c9-4152-a9c2-bd702058cde6)

**Comment by mforell** [comment]
We were able to make it work by not using the built-in editor but instead creating it locally and pushing the template to the instance with Coder. I believe this example demonstrates the process well (https://github.com/matifali/coder-templates/tree/main/deeplearning-nvidia). In the "images" folder, we have, for instance, a requirements.txt file that we use in the Dockerfile.

**Comment by jizusun** [comment]
Hi @mforell  I don't see a `requirements.txt` in your referenced link

![image](https://github.com/coder/coder/assets/4011348/73f2b7ab-3cae-4512-8b77-3760077213c4)


Do you mean another link?

**Comment by mforell** [comment, marked as answer]
Hi, sorry for the confusion.
I just made a sample project for you:
https://github.com/mforell/coder-template-example 

You'll just have to copy the files you need in the Dockerfile. 

In the tensorflow.Dockerfile line 42 the requirements.txt is copied and line 60 e.g. a main.py
